haystack/preview/document_stores/memory/document_store.py

```diff
@@ -259,3 +270,106 @@ def bm25_retrieval(
             return_document = Document(**doc_fields)
             return_documents.append(return_document)
         return return_documents
+
+    def embedding_retrieval(
+        self,
+        query_embedding: List[float],
+        filters: Optional[Dict[str, Any]] = None,
+        top_k: int = 10,
+        scale_score: bool = True,
```

Comment:
I am not sure that `scale_score` should be True by default.
My impression is that the current scaling factors are a bit arbitrary and can introduce unintended effects.

[Related Discord thread](https://discord.com/channels/993534733298450452/1124729047906267258)

Reply:
@julian-risch @sjrl Any ideas on this point?

Reply:
Having scaling activated by default makes sense to me. For first time users it's more intuitive and that way the range of scores doesn't change when switching document stores/models. The current default in 1.x is `True` too.
However, you're completely right that it can lead to unexpected results and scaling is only on a "best effort" basis. Having beginners in mind scaling is still the better option I believe and advanced users can decide on their own.
From the user perspective I wonder why `scale_score` is a parameter of `MemoryDocumentStore`. Why not of `MemoryEmbeddingRetriever` instead? How would we explain that?

Reply:
Thanks @julian-risch!
Your observation makes sense to me -> I will set `scale_score` default value to True.

`scale_score` is a parameter of the `embedding_retrieval` method of `MemoryDocumentStore`, which is invoked by the `MemoryEmbeddingRetriever`.
See for example the future `init` and `run` methods of the `MemoryEmbeddingRetriever`:
```python
    def __init__(
        self,
        document_store: MemoryDocumentStore,
        filters: Optional[Dict[str, Any]] = None,
        top_k: int = 10,
        scale_score: bool = True,
    ):
    ...
    
    @component.output_types(documents=List[List[Document]])
    def run(
        self,
        queries_embeddings: List[List[float]],
        filters: Optional[Dict[str, Any]] = None,
        top_k: Optional[int] = None,
        scale_score: Optional[bool] = None,
        ...
```
So, this parameter can be set and changed in the Retriever...